src/Factory/ProductListViewFactory.php

```diff
@@ -101,14 +96,19 @@ private function getImageViews(Collection $images)
 
     /**
      * @param Collection|TaxonDocument[] $taxons
+     * @param TaxonDocument|null $mainTaxonDocument
      *
-     * @return TaxonView[]
+     * @return array
      */
-    private function getTaxonViews(Collection $taxons)
+    private function getTaxonCodes(Collection $taxons, TaxonDocument $mainTaxonDocument = null)
     {
         $taxonViews = [];
+        if (null !== $mainTaxonDocument) {
+            $taxonViews['main'] = $mainTaxonDocument->getCode();
+        }
+
         foreach ($taxons as $taxon) {
-            $taxonViews[] = $this->getTaxonView($taxon);
+            $taxonViews['others'][] = $taxon->getCode();
         }
 
         return $taxonViews;
```

Comment:
It would return an empty array if no taxons are passed, but `main` and `others` keys should always exist (with `null` and `[]` values respectively).